libmamba/include/mamba/core/util_string.hpp

```diff
@@ -21,108 +21,282 @@
 
 namespace mamba
 {
-    inline const char* check_char(const char* ptr)
+    /**
+     * Return the string if the pointer is not null, otherwise a pointer to an empty string.
+     */
+    const char* raw_str_or_empty(const char* ptr);
+
+    char to_lower(char c);
+    wchar_t to_lower(wchar_t c);
+    std::string to_lower(std::string_view str);
+    std::wstring to_lower(std::wstring_view str);
+    template <typename Char>
+    std::basic_string<Char> to_lower(std::basic_string<Char>&& str);
+    template <typename Char, typename Iter>
+    void to_lower_tranform(std::basic_string_view<Char> str, Iter out);
```

Comment:
Typo. Also it would be more C++ish to have a full iterator interface.
And design question: Do we want to keep a template function for `to_lower` and `to_lower_transform` in the API? We could also have it as an implementation detail and a restricted API on `string` and `wstring` like we have for `string_view` and `wstring_view`. Or we could have a global template for handling these 4 use cases.